lib/manager/gomod/artifacts.ts

```diff
@@ -41,53 +41,42 @@ export async function updateArtifacts(
     const customEnv = ['GOPATH', 'GOPROXY', 'GONOSUMDB'];
     const env = getChildProcessEnv(customEnv);
     const startTime = process.hrtime();
-    let cmd: string;
+    const cmd = 'go';
+    const execOptions: ExecOptions = { cwd, env };
     if (config.binarySource === 'docker') {
       logger.info('Running go via docker');
-      cmd = `docker run --rm `;
-      if (config.dockerUser) {
-        cmd += `--user=${config.dockerUser} `;
-      }
-      const volumes = [config.localDir, process.env.GOPATH];
-      cmd += volumes.map(v => `-v "${v}":"${v}" `).join('');
-      const envVars = customEnv;
-      cmd += envVars.map(e => `-e ${e} `).join('');
-      cmd += '-e CGO_ENABLED=0 ';
-      cmd += `-w "${cwd}" `;
-      cmd += `renovate/go `;
       const credentials = find({
         hostType: 'github',
         url: 'https://api.github.com/',
       });
-      if (credentials && credentials.token) {
-        logger.debug('Setting github.com credentials');
-        cmd += `bash -c "git config --global url.\\"https://${
-          global.appMode
-            ? `x-access-token:${credentials.token}`
-            : credentials.token
-        }@github.com/\\".insteadOf \\"https://github.com/\\" && go`;
-      } else {
-        cmd += 'go';
-      }
+      const cmdWrap =
+        credentials && credentials.token
+          ? `bash -c "git config --global url.\\"https://${
```

Comment:
Do you think the "bash -c" part could be abstracted too? Really it only exists so that we can run more than one command at a time and is not manager-specific. We could for example have arrays of "dockerPreCommands" and "dockerPostCommands" and if they are non-empty then we concatenate all commands together with `&&` and wrap with `bash -c ""`

Reply:
Instead of `cmdWrap`

Reply:
Pre /post commands would be very useful